api/src/services/import-services/observation/utils/observation-sampling-row-validator.ts

```diff
@@ -0,0 +1,446 @@
+import { CSVConfigUtils } from '../../../../utils/csv-utils/csv-config-utils';
+import { CSVRowValidator } from '../../../../utils/csv-utils/csv-config-validation.interface';
+import { updateCSVRowState } from '../../../../utils/csv-utils/csv-header-configs';
+import { formatDateString, isDateString, isDateTimeString, isTimeString } from '../../../../utils/date-time-utils';
+import { ObservationCSVStaticHeader } from '../import-observations-service';
+
+dayjs.extend(isSameOrAfter);
+dayjs.extend(isSameOrBefore);
+
+/**
+ * Observation Sampling Information Row Validator - This function will validate a row of observation data and ensure
+ * that the sampling information provided in the row matches a valid sampling period from the provided list of
+ * sampling periods.
+ *
+ * Successfull paths:
+ *  1. No sampling information provided, but observation date / time is provided (no period needed)
+ *  2. Exact period match found using site, technique, and period
+ *  3. Multiple periods match the site, technique, and period, but only one matches the observation date and time
+ *
+ * @param {SurveySamplePeriodDetails[]} samplingPeriods All available sampling periods for the survey.
+ * @param {CSVConfigUtils<ObservationCSVStaticHeader>} utils CSV Config Utils for the observation CSV.
+ * @return {*} {CSVRowValidator}
+ */
+export function getObservationSamplingInformationRowValidator(
+  samplingPeriods: SurveySamplePeriodDetails[],
+  utils: CSVConfigUtils<ObservationCSVStaticHeader>
+): CSVRowValidator {
+  return (params) => {
+    // Extract site, technique, and period data from the row
+    const worksheetSiteName = utils.getCellValue('SAMPLING_SITE', params.row) as string | null;
+    const worksheetTechniqueName = utils.getCellValue('METHOD_TECHNIQUE', params.row) as string | null;
+    const worksheetPeriod = utils.getCellValue('SAMPLING_PERIOD', params.row) as string | null;
+
+    // Extract observation date and time from the row
+    const worksheetObservationDate = utils.getCellValue('DATE', params.row) as string | null;
+    const worksheetObservationTime = utils.getCellValue('TIME', params.row) as string | null;
+
+    // Determine if the worksheet contains any sampling information
+    const worksheetHasSamplingInformation = worksheetSiteName || worksheetTechniqueName || worksheetPeriod;
+
+    // VALID: No sampling information provided, but observation date / time is provided (no period needed)
+    if (!worksheetHasSamplingInformation && worksheetObservationDate) {
+      return [];
+    }
+
+    // INVALID: No sampling information or observation date / time provided
+    if (!worksheetHasSamplingInformation && !worksheetObservationDate) {
+      return [
+        {
+          error: 'Row does not contain sampling information or observation date / time',
+          solution: 'Please provide sampling information or a valid observation date and time',
+          header: null,
+          cell: null
+        }
+      ];
+    }
```

Comment:
Does this need to check for lat/long as well?

Reply:
I actually handle this case in the `observation-location-row-validator.ts`

Reply:
I wonder if the error messages from this validator and the lat/lng validator should be more aligned, just because they are doing very similar checks regarding defined vs undefined sampling.

`Latitude is required when sampling information is not provided`
`Please provide sampling information or a valid latitude`

`Longitude is required when sampling information is not provided`
`Please provide sampling information or a valid longitude`

`Date / time is required when sampling information is not provided`
`Please provide sampling information or a valid observation date / time`

Reply:
Yea I was uncertain if it was better to have this all combined or not. Mostly have them seperate as it was an afterthought tbh. I can combine them and drop the other validator.

Reply:
I dont think separate validators is necessarily an issue. I mostly just mean having the error message written in a similar way, whether its across multiple validators or not.